src/main/java/org/ow2/proactive/workflow_catalog/rest/util/ArchiveManagerHelper.java

```diff
@@ -0,0 +1,137 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.ow2.proactive.workflow_catalog.rest.entity.WorkflowRevision;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class ArchiveManagerHelper {
+
+    /**
+     * Get the name of the Workflow used into the archive (in order not to have twice the same
+     * name). If the name of the Workflow already exists, an index is append to the name.
+     * @param existingNames the set of names already used
+     * @param currentName the Workflow current name
+     * @return the name of the Workflow different from all existing names
+     */
+    public String getName(Set<String> existingNames, String currentName) {
+
+        if (!existingNames.contains(currentName))
+            return currentName;
+
+        for (int i = 1; i <= existingNames.size(); i++) {
+            String newName = currentName + "_" + i;
+            if (!existingNames.contains(newName)) {
+                return newName;
+            }
+        }
+
+        return currentName + "_0";
+
+    }
+
+    /**
+     * Compress a list of WorkflowRevision into a ZIP archive
+     * @param workflowsList the list of workflows to compress
+     * @return a byte array corresponding to the archive containing the workflows
+     */
+    public byte[] compressZIP(List<WorkflowRevision> workflowsList) {
+
+        if (workflowsList == null || workflowsList.size() == 0) {
+            return null;
+        }
+
+        try {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream);
+
+            Set<String> existingNames = new HashSet<>();
+            for (WorkflowRevision file : workflowsList) {
+                String fileName = getName(existingNames, file.getName());
+                existingNames.add(fileName);
+
+                byte[] xmlByteArray = file.getXmlPayload();
+
+                ZipEntry entry = new ZipEntry(fileName);
+                entry.setSize(xmlByteArray.length);
+
+                zipOutputStream.putNextEntry(entry);
+                zipOutputStream.write(xmlByteArray);
+                zipOutputStream.closeEntry();
+            }
+
+            zipOutputStream.close();
+            return byteArrayOutputStream.toByteArray();
+        } catch (IOException ioe) {
+            throw new RuntimeException(ioe);
+        }
+
+    }
+
+    /**
+     * Extract workflows from an archive
+     * @param byteArrayArchive the archive as byte array
+     * @return the list of Workflows byte arrays
+     */
+    public List<byte[]> extractZIP(byte[] byteArrayArchive) {
+
+        List<byte[]> filesList = new ArrayList<>();
```

Comment:
@lpellegr you mean initialise the capacity as 0 ? from java8, it is not necessary anymore. it is 0 by default.